system-ballerina/build.gradle

```
@@ -128,7 +128,38 @@ task revertTomlFile {
     }
 }
 
+task ballerinaTest {
+    dependsOn(copyStdlibs)
+    dependsOn(":system-native:build")
+    dependsOn(":system-test-utils:build")
+    dependsOn(updateTomlVerions)
+    finalizedBy(revertTomlFile)
+
+    def distributionBinPath =  project.projectDir.absolutePath + "/build/target/extracted-distributions/jballerina-tools-zip/jballerina-tools-${ballerinaLangVersion}/bin"
+    def debugParams = ""
+    if (project.hasProperty("debug")) {
+        debugParams = "--debug ${project.findProperty("debug")}"
+    }
+
+    doLast {
+        exec {
+            workingDir project.projectDir
+            environment "JAVA_OPTS", "-DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true"
+            if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                commandLine 'cmd', '/c', "$distributionBinPath/ballerina.bat test --code-coverage --all ${debugParams}"
+            } else {
+                commandLine 'sh', '-c', "$distributionBinPath/ballerina test --code-coverage --all ${debugParams}"
+            }
+        }
+    }
+}
+
+test {
+    dependsOn(ballerinaTest)
+}
+
 task ballerinaBuild {
+    dependsOn(test)
```

```suggestion
    dependsOn(copyStdlibs)
    dependsOn(":system-native:build")
    dependsOn(test)
```
Add these dependency too. Else if you skip these will also not run.